### ISSUE TITLE
ImportError: Could not import PIL.Image.

### ISSUE THREAD

**Issue by WuyangLI** [aws/sagemaker-python-sdk#231]
### System Information
- **Framework (e.g. TensorFlow) / Algorithm (e.g. KMeans)**: tensorflow, keras
- **Framework Version**: 1.5.0
- **Python Version**: 2.7
- **CPU or GPU**: GPU
- **Python SDK Version**: 1.4.1
- **Are you using a custom image**: No

### Describe the problem
Pillow installed using requirements.txt is not recognized by Keras.
Keras ImageDataGenerator throws the following exception:
UnknownError: exceptions.ImportError: Could not import PIL.Image. The use of `array_to_img` requires PIL.
This issue may be related to issue #84  Could Pillow also be included in the docker image (as done with h5py)?

### Minimal repro / logs

**keras_pillow.py** 

```
import tensorflow as tf

INPUT_TENSOR_NAME = "input_1"
NUM_CLASSES = 2
BATCH_SIZE = 10


def keras_model_fn(hyperparameters):
    model = tf.keras.applications.vgg16.VGG16(include_top=True, weights=None, classes=NUM_CLASSES)
    model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    return model


def _input_fn(training_dir, input_shape, batch_size):
    preprocess_input = tf.keras.applications.vgg16.preprocess_input

    generator = tf.keras.preprocessing.image.ImageDataGenerator(preprocessing_function=preprocess_input)\
        .flow_from_directory(training_dir, target_size=input_shape, batch_size=batch_size)

    tensor_shapes = (tf.TensorShape([None, input_shape[0], input_shape[1], 3]), tf.TensorShape([None, NUM_CLASSES]))
    tensor_types = (tf.float32, tf.float32)
    dataset = tf.data.Dataset.from_generator(lambda: generator, tensor_types, tensor_shapes)
    features, labels = dataset.make_one_shot_iterator().get_next()
    return {INPUT_TENSOR_NAME: features}, labels


def train_input_fn(training_dir, hyperparameters):
    return _input_fn(training_dir + '/train/', (224, 224), BATCH_SIZE)


def eval_input_fn(training_dir, hyperparameters):
    return _input_fn(training_dir + '/test/', (224, 224), BATCH_SIZE)


def serving_input_fn(hyperparameters):
    inputs = {INPUT_TENSOR_NAME: tf.placeholder(tf.float32, [None, 224, 224, 3])}
    return tf.estimator.export.ServingInputReceiver(inputs, inputs)
```

**requirements.txt**

```
Pillow
```

**Training job**

```
import sagemaker

from sagemaker.tensorflow import TensorFlow
from sagemaker.session import s3_input
from sagemaker import get_execution_role

sagemaker_session = sagemaker.Session()
role = get_execution_role()
training_steps = 100
evaluation_steps = 10

estimator = TensorFlow(
    entry_point='keras_pillow.py',
    source_dir='./',
    role=role,
    training_steps=100,
    evaluation_steps=10,
    train_instance_count=1,
    train_instance_type='ml.p2.xlarge',
    requirements_file='requirements.txt',
    input_mode='File')

input_dataset = s3_input('s3://com.amazon.evi.multimedia.ingestion/temp/ocorcoll/datasets/cats_and_dogs/')
estimator.fit(input_dataset)
```

This produces the following logs:

```
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 905, in run
    run_metadata_ptr)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 1137, in _run
    feed_dict_tensor, options, run_metadata)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 1355, in _do_run
    options, run_metadata)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/client/session.py", line 1374, in _do_call
    raise type(e)(node_def, op, message)
UnknownError: exceptions.ImportError: Could not import PIL.Image. The use of `array_to_img` requires PIL.
#011 [[Node: PyFunc = PyFunc[Tin=[DT_INT64], Tout=[DT_FLOAT, DT_FLOAT], token="pyfunc_1"](arg0)]]
#011 [[Node: IteratorGetNext = IteratorGetNext[output_shapes=[[?,224,224,3], [?,2]], output_types=[DT_FLOAT, DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](OneShotIterator)]]
#011 [[Node: training/Adam/gradients/loss/predictions_loss/mul_grad/Shape/_833 = _HostRecv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_443_training/Adam/gradients/loss/predictions_loss/mul_grad/Shape", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
```

**Comment by laurenyu** [comment]
Thanks for all the detail! It does indeed look like pillow needs to be imported before tensorflow is. I'll try to get to it soon, though you're also welcome to beat me to it and send a PR in our TensorFlow container repo: https://github.com/aws/sagemaker-tensorflow-containers

**Comment by WuyangLI** [comment]
Hi, Lauren, thank you. I'll make a pull request at the end of this week :).
best regards.

Wuyang

**Comment by laurenyu** [comment]
hi @WuyangLI, oops, probably should've updated here - I've just updated a PR to add Pillow: aws/sagemaker-tensorflow-containers#48

**Comment by laurenyu** [comment]
This change has now been released for TF versions 1.5 and higher.
